erpnext/selling/report/customer_acquisition_and_loyalty/customer_acquisition_and_loyalty.py

```diff
@@ -2,65 +2,201 @@
+                'total_revenue': new[1] + repeat[1]
+            })
+    return columns, out
+
+def get_data_by_territory(filters, common_columns):
+    columns = [{
+        'label': 'Territory',
+        'fieldname': 'territory',
+        'fieldtype': 'Link',
+        'options': 'Territory',
+        'width': 150
+    }]
+    columns += common_columns
+
+    customers_in = get_customer_stats(filters, tree_view=True)
+
+    territory_dict = {}
+    for t in frappe.db.sql('''SELECT name, lft, parent_territory, is_group FROM `tabTerritory` ORDER BY lft''', as_dict=1):
+        territory_dict.update({
+            t.name: {
+                'parent': t.parent_territory,
+                'is_group': t.is_group
+            }
+        })
+
+    depth_map = frappe._dict()
+    for name, info in territory_dict.items():
+        default = depth_map.get(info['parent']) + 1 if info['parent'] else 0
+        depth_map.setdefault(name, default)
+
+    data = []
+    for name, indent in depth_map.items():
+        condition = customers_in.get(name)
+        new = customers_in[name]['new'] if condition else [0, 0.0]
+        repeat = customers_in[name]['repeat'] if condition else [0, 0.0]
+        temp = {
+            'territory': name,
+            'indent': indent,
+            'new_customers': new[0],
+            'repeat_customers': repeat[0],
+            'total': new[0] + repeat[0],
+            'new_customer_revenue': new[1],
+            'repeat_customer_revenue': repeat[1],
+            'total_revenue': new[1] + repeat[1],
+            'bold': 0 if indent else 1
+        }
+        data.append(temp)
+    node_list = [x for x in territory_dict.keys() if territory_dict[x]['is_group'] == 0]
+    root_node = [x for x in territory_dict.keys() if territory_dict[x]['parent'] is None][0]
+    root_data = [x for x in data if x['territory'] == root_node][0]
+
+    for node in node_list:
+        data = update_groups(node, data, root_node, territory_dict)
+
+    for group in [x for x in territory_dict.keys() if territory_dict[x]['parent'] == root_node]:
+        group_data = [x for x in data if x['territory'] == group][0]
+        for key in group_data.keys():
+            if key not in ['indent', 'territory', 'bold']:
+                root_data[key] += group_data[key]
+
+    return columns, data, None, None, None, 1
+
+def update_groups(node, data, root_node, territory_dict):
+    """ Adds values of child territories to parent node except root. """
+    parent_node = territory_dict[node]['parent']
+    if parent_node != root_node and parent_node:
+        node_data = [x for x in data if x['territory'] == node][0]
+        parent_data = [x for x in data if x['territory'] == parent_node][0]
+        for key in parent_data.keys():
+            if key not in ['indent', 'territory', 'bold']:
+                parent_data[key] += node_data[key]
+        return update_groups(parent_node, data, root_node, territory_dict)
+    else:
+        return data
```

Comment:
No need of all this, why not update the update parent territories in the above for loop itself?

Reply:
Because, when focus is on node and you update the groups from the node values, the root gets that value added twice, once from the node and then once when the group comes in focus in the loop due to recursion.
Hence a separate loop to sum the group values to the root...

It's tested with all possibilities, works fine...